code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -345,25 +390,35 @@ func (sc *StorageSmartContract) newAllocationRequestInternal(
 	}
 
 	// create write pool and lock tokens
-	if err := sa.addToWritePool(txn, balances, transfer); err != nil {
+	if err := alloc.addToWritePool(txn, balances, transfer); err != nil {
 		logging.Logger.Error("new_allocation_request_failed: error adding to allocation write pool",
 			zap.String("txn", txn.Hash),
 			zap.Error(err))
 		return "", common.NewError("allocation_creation_failed", err.Error())
 	}
 
-	if err := sa.checkFunding(); err != nil {
+	if err := alloc.checkFunding(); err != nil {
 		return "", common.NewError("allocation_creation_failed", err.Error())
 	}
 	m.tick("create_write_pool")
 
-	if err = sc.createChallengePool(txn, sa, balances, conf); err != nil {
-		logging.Logger.Error("new_allocation_request_failed: error creating challenge pool",
-			zap.String("txn", txn.Hash),
-			zap.Error(err))
-		return "", common.NewError("allocation_creation_failed", err.Error())
+	if !request.IsEnterprise {
+		if err = sc.createChallengePool(txn, alloc, balances, conf); err != nil {
```

Comment:
We need activation to switch off the createChallengePool, imagine if both old and new miners received the request with `IsEnterprise=true`, the old one will always create the challenge pool, while the new one will skip.

Reply:
nvm, seems fixed in PR https://github.com/0chain/0chain/pull/3357